### ISSUE TITLE
Would you be interested in training a small foundational model with Sophia?

### ISSUE THREAD

**Issue by Mythli** [Liuhong99/Sophia#15]
Hi!

Am I right in assuming, that Sophia seems to plateau later than Adam? Training on a large amount of tokens could therefore result in a very small and capable model with state of the art performance.  

If so, would you be interested in training a small (3-7b) foundational model using the Sophia optimizer on a large amount of tokens (1-2T)? 

I can sponsor the compute.

If you are interested, just drop me a short note or let's have a short video chat.

You can write me under tobias at web-bee.org
Or Discord mythli#5017

Feel free to book any time in my calendar that fits your schedule: https://calendar.app.google/cD59ASVPtWVEFFqy8

Best regards from Germany
Tobias

**Comment by Liuhong99** [comment]
Hi Tobias,

Thanks for the interest in Sophia and sorry for my late reply! 

> Sophia seems to plateau later than Adam? 

This is generally correct if we use the same weight decay as Adam.

> would you be interested in training a small (3-7b) foundational model using the Sophia optimizer on a large amount of tokens (1-2T)?

I'm happy to! I'm running 1.5B models with Sophia and the progress seems great so far. Looking forward to chatting with you about details.

Best,
Hong

**Comment by jzhang38** [comment]
@Liuhong99 Could you share the hyperparameters you used for the 1.5B model?

**Comment by Liuhong99** [comment]
Sophia: lr = 2e-4, gamma = 0.04, wd = 0.15, beta1 = 0.965, beta2 = 0.99
Adam: lr = 1.2e-4, wd = 0.1, beta1 = 0.9, beta2 = 0.95
I'm using another code base of the 1.5B models, which will be released soon. gamma can depend on the initialization and architecture details of the code base. A general recipe is to make the win_rate tracked in train_sophiag.py about 30-40%
